moped-database/migrations/1700149643858_parent_audit_values/up.sql

```diff
@@ -0,0 +1,73 @@
+-- Altering moped_project table to add a new column for tracking the user who last updated a record
+ALTER TABLE moped_project ADD COLUMN updated_by_user_id INTEGER DEFAULT NULL;
+COMMENT ON COLUMN moped_project.updated_by_user_id IS 'User ID of the person who last updated the project';
+
+-- Creating or replacing a function to update audit logs in parent records
+CREATE OR REPLACE FUNCTION update_parent_records_audit_logs()
+RETURNS TRIGGER AS $$
+DECLARE
+  project_id_variable INTEGER;
+  user_id_variable INTEGER;
+  query TEXT;
+BEGIN
+  
+  -- Constructing a dynamic query to select the user ID from the triggered table
+  query := 'SELECT updated_by_user_id FROM ' || quote_ident(TG_TABLE_NAME) || 
+           ' WHERE id' || ' = ' || NEW.id || ';';
+  
+  -- Executing the dynamic query
+  EXECUTE query INTO user_id_variable;
+
+  -- Update the project component with the current timestamp and user ID
+  UPDATE moped_proj_components
+  SET updated_at = NOW(), updated_by_user_id = user_id_variable
```

Comment:
i wonder if it might be better to use the `NEW.updated_at` value, so that we have identical `updated_at` timestamps across all the affected records. i know we're just talking about microseconds here 🙈

Reply:
Absolutely. There's no reason in the world not to align them and that change is made in here: 7972d8b9a3a83ba5638e273bb4ce06d299f48890.